platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/GossipEvent.java

```diff
@@ -83,6 +113,10 @@ public void deserialize(final SerializableDataInputStream in, final int version)
             in.readLong(); // roundCreated
         }
         timeReceived = Instant.now();
+
+        if (hashedData == null || unhashedData == null || !isHashedDataInternallyConsistent(hashedData)) {
+            throw new IOException("Invalid event: " + this);
+        }
```

Comment:
It would be nice if the message included the reason why the event is invalid. Not very useful if an attacker is sending a bad event, but perhaps helpful when debugging.

Reply:
Logging improved